### ISSUE TITLE
Silent removal of Wazuh cluster synchronization of agent.conf files in 3.2.3?

### ISSUE THREAD

**Issue by branchnetconsulting** [wazuh/wazuh#691]
I just upgraded my Wazuh cluster from 3.2.2 to 3.2.3 and it appears now the agent.conf files on my master manager node no longer are getting synced to the client manager nodes.  What happened?  I see no reference to the removal of this valuable feature in the change log, release notes, or in the manual for 3.2.3.  The only history I can find on this is that this line was removed in a 3.2.3 documentation update in the area addressing Wazuh clusters:

`-- centralized configuration of the ``agent.conf`` file used by the agents within each agent group.`

I believe syncing of agent.conf across the Wazuh manager cluster has been a feature since 3.0.0.  Where did it go and why?  

Kevin

**Comment by mgmacias95** [comment]
Hello @branchnetconsulting,

We have stopped synchronizing all group files, now **we only synchronize the `merged.mg` file of each group**. 

The `merged.mg` file is a file where all group files are merged into a single one, including the `agent.conf`. 

Don't worry about this, the agent will be still receiving its configuration just like it did in previous versions.

Best regards,
Marta

**Comment by branchnetconsulting** [comment]
Marta, 

Thanks for clarifying this.  In my case, my **merged.mg** file was not being recreated upon restart of the master-node wazuh-manager.   Thanks to your and Jesus' help, it became evident that this was due to me having commented out the entire `<remote>` section in **ossec.conf** on the master node.  I did that because I do not want to accept Wazuh agent connections on my master node, and assumed I had no need for the ossec-remoted daemon running at all.  However, it turns out ossec-remoted is responsible for generating **merged.mg**, which as of 3.2.3 I very much am depending on for propagating group file changes from master node to client nodes.  After putting `<remote>` back to normal, things starting working for me.  My ufw firewall on the master node is already not allowing Wazuh agent connections anyway.  If I were not using a local firewall, I would just bind ossec-remoted to listen only on 127.0.0.1 as explained here:

https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/remote.html#local-ip

On a related note, now that for agent group files, only **merged.mg** is being sent from master node to the client nodes, it is harder to understand and debug the propagation of group file changes across the manager cluster.  Thus I am proposing in #693 that we have the client nodes unpack the **merged.mg** files back into individual group files so we can see and debug the process more easily.

Thanks for your help!
Kevin Branch
